src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt

```diff
@@ -78,10 +77,10 @@ class CargoWorkspace private constructor(
 
     private val targetByCrateRootUrl = packages.flatMap { it.targets }.associateBy { it.crateRootUrl }
 
-    fun findCrateByName(normName: String): Target? =
-        packages
-            .mapNotNull { it.libTarget }
-            .find { it.normName == normName }
+    fun findCrateByName(normName: String, inPackage: Package? = null): Target? =
```

Comment:
Can we make this last argument non-optional (or perhaps make this function a method of `Package`)? Current signature is (and always was) a bit of a lie, because if you don't know the package, you might get wrong results.

If we indeed sometime need to find crate by name "approximately", it's better to create a separate function with `approximately` or something in the name :)